lib/azure/core/http/http_error.rb

```diff
@@ -80,24 +81,12 @@ def initialize(http_response)
         #
         # Returns nothing
         def parse_response
-          if @http_response.body && @http_response.body.include?('<')
-
-            document = Nokogiri.Slop(@http_response.body)
-
-            @type = document.css('code').first.text if document.css('code').any?
-            @type = document.css('Code').first.text if document.css('Code').any?
-            @description = document.css('message').first.text if document.css('message').any?
-            @description = document.css('Message').first.text if document.css('Message').any?
-
-            @header = document.css('headername').first.text if document.css('headername').any?
-            @header = document.css('HeaderName').first.text if document.css('HeaderName').any?
-
-            @header_value = document.css('headervalue').first.text if document.css('headervalue').any?
-            @header_value = document.css('HeaderValue').first.text if document.css('HeaderValue').any?
-
-            # service bus uses detail instead of message
-            @detail = document.css('detail').first.text if document.css('detail').any?
-            @detail = document.css('Detail').first.text if document.css('Detail').any?
+          if @http_response.body
+            if @http_response.body.include?('<')
```

Comment:
I understand it is from previous code, but it is brittle. Json body can contain the `<` character. Does it make sense to check the starting and ending character to make the first judgement whether it is XML or Json?

Reply:
Actually, it is better we use the `Content-Type` header instead, as we talked offline and confirmed that if the response has body, a `Content-Type` header should exist. The updated PR reflects the change.